server/src/main/java/org/apache/druid/server/coordinator/ReservoirSegmentSampler.java

```diff
@@ -31,6 +32,42 @@
 
   private static final EmittingLogger log = new EmittingLogger(ReservoirSegmentSampler.class);
 
+  static List<BalancerSegmentHolder> getRandomBalancerSegmentHolders(
+      final List<ServerHolder> serverHolders,
+      Set<String> broadcastDatasources,
+      int k
```

Comment:
does it make sense to retain the `percentOfSegmentsToConsider` functionality to allow short-circuiting iterateAllSegments across all servers? We update the docs https://github.com/apache/druid/blob/master/docs/configuration/index.md#dynamic-configuration accordingly either way

Reply:
It makes sense to me to retain the `percentOfSegmentsToConsider` and add an another dynamic parameter `useBatchedSegmentSampler` to enable the new improvement. In this way, the changing will be less aggressive.